platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -0,0 +1,231 @@
+# Lifecycle management for the State
+
+## Summary
+
+This proposal describes a possible implementation of state lifecycle management that could work equally well for both the Consensus Node and the Block Node.
+
+|      Metadata      |                 Entities                 |
+|--------------------|------------------------------------------|
+| Designers          | [@imalygin](https://github.com/imalygin) |
+| Functional Impacts | Consensus Node, Block Node               |
+
+## Purpose and Context
+
+There is an initiative to create self-contained State modules (`swirlds-state-api` and `swirlds-state-impl`) that would have minimal dependencies, and specifically, they should not depend on `swirlds-platform-core`. Part of this initiative is the implementation of a state lifecycle management mechanism that could be used by both the Consensus Node and the Block Node.
+
+### Requirements
+
+- Add an interface representing state lifecycle management and its implementation.
+- Update the current State API to be coherent with this State Lifecycle API.
+- Implement this set of APIs and migrate the existing code to use it.
+
+## Design & Architecture
+
+### Reservation count mechanism
+
+This mechanism is already implemented as a part of the Merkle tree implementation. However, it's importnant to mention it here
+to provide important context for understanding how exactly the state lifecycle management will work.
+
+Every Merkle tree node implements the following interface:
+
+```java
```

Comment:
I think putting this code block here is not necessary, and is in fact distracting. It's large and takes up a lot of space, and breaks up my mental flow. I think a few sentences describing the basics of the reservations would be much better.

Reply:
Fair enough. I'll remove the code and replace it with the summary of the idea.

Reply:
Done.